tests/components/holiday/test_init.py

```diff
@@ -31,7 +29,9 @@ async def test_unload_entry(hass: HomeAssistant) -> None:
     await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
 
+    assert entry.state == ConfigEntryState.LOADED
+
     assert await hass.config_entries.async_unload(entry.entry_id)
     await hass.async_block_till_done()
 
-    assert entry.state is ConfigEntryState.NOT_LOADED
+    assert entry.state is not ConfigEntryState.LOADED
```

Comment:
Why did this change?

Reply:
Mypy complained (`Non-overlapping identity check (left operand type: "Literal[ConfigEntryState.LOADED]", right operand type: "Literal[ConfigEntryState.NOT_LOADED]")` and I did not find a solution other than that.

Reply:
I'm not a typing expert but is it because one check is using `==` and one is using `is`?

I'm just looking at some recent examples and I don't see a problem checking explicitly for `NOT_LOADED`
https://github.com/home-assistant/core/blob/75f237b587b2912e07af896ea114d59f5d403b55/tests/components/fitbit/test_init.py#L37
or 
https://github.com/home-assistant/core/blob/75f237b587b2912e07af896ea114d59f5d403b55/tests/components/gios/test_init.py#L55

Reply:
No difference between `is` and `==`

Reply:
OK maybe it is because they didn't enable strict typing. I see this discussed here https://github.com/python/mypy/issues/9005

I think it would be better to disable the type check with a comment, or workaround the type checking some other way (looking up the config entry state like other tests do) than not checking for the state explicitly just in case it is some other enum value (e.g. `SETUP_ERROR`)

Reply:
I was pretty certain we didn't do mypy on tests?

Reply:
@jrieger can you make a comment in the code referring to the mypy issue @allenporter linked too? Thanks

Reply:
Done.